Refactors landmines

Thunder12345 opened the pull request:
Quick refactor to make landmine code more readable. Gonna be using these for an away mission I'm working on, so I decided to clean them up before using them.

Commit:
Refactors landmines

Replaced the minepayload var and the switch() in triggermine() with a
proc for the mine's effect.
Removed unnessecary New() proc.
Removed the existing uglymine icon state and renamed uglyminearmed to
uglymine.

phil235 commented on code/game/objects/effects/mines.dm:
```diff
@@ -24,48 +23,42 @@
+
+/obj/effect/mine/kickmine/mineEffect(mob/victim)
+	if(isliving(victim) && victim.client)
+		victim << "<font color='red'><b>You have been kicked FOR NO REISIN!<b></font>"
```
span class = userdanger please.

Commit:
Now with more (span) class

(user)DANGER (user)DANGER!

Miauw62 commented on code/game/objects/effects/mines.dm:
```diff
@@ -24,48 +23,42 @@
+
+/obj/effect/mine/stun/mineEffect(mob/victim)
+	if(isliving(victim))
+		victim.Weaken(8)
```
Fairly sure this is 8 ticks, not 8 seconds.

Iamgoofball replied:
actually, human life triggers every 2.1 seconds

so this is 16 seconds.

Miauw62 commented:
you might want to add an "effect" var to these landmines and do victim.Weaken(8 * effect) or something like that for easily making more/less powerful mines.

Commit:
More landmine variabl-HONK

Gave mines more vars related to their effects, repathed some mines to
make this easier, and added new sound mines, that play a sound when
triggered. Who let that clown in?

Thunder12345 commented:
Travis why have you abandoned me
